apps/omg_eth/lib/omg_eth/ethereum_height_monitor.ex

```diff
@@ -0,0 +1,204 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Eth.EthereumHeightMonitor do
+  @moduledoc """
+  Periodically calls the Ethereum client node to check for Ethereumm's block height. Publishes
+  internal events or raises alarms accordingly.
+
+  When a new block height is received, it publishes an internal event under the topic `"ethereum_new_height"`
+  with the payload `{:ethereum_new_height, height}`. The event is only published when the received
+  block height is higher than the previously published height.
+
+  When the call to the Ethereum client fails or returns an invalid responnse, it raises an
+  `:ethereum_connection_error` alarm. The alarm is cleared once a valid block height is seen.
+
+  When the call to the Ethereum client returns the same block height for longer than
+  `:ethereum_stalled_sync_threshold_ms`, it raises an `:ethereum_stalled_sync` alarm.
+  The alarm is cleared once the block height starts increasing again.
+  """
+  use GenServer
+  require Logger
+
+  @type t() :: %__MODULE__{
+          check_interval_ms: pos_integer(),
+          stall_threshold_ms: pos_integer(),
+          tref: reference() | nil,
+          eth_module: module(),
+          alarm_module: module(),
+          event_bus: module(),
+          ethereum_height: integer(),
+          synced_at: DateTime.t(),
+          connection_alarm_raised: boolean(),
+          stall_alarm_raised: boolean()
+        }
+
+  defstruct check_interval_ms: 10_000,
+            stall_threshold_ms: 20_000,
+            tref: nil,
+            eth_module: nil,
+            alarm_module: nil,
+            event_bus: nil,
+            ethereum_height: 0,
+            synced_at: nil,
+            connection_alarm_raised: false,
+            stall_alarm_raised: false
+
+  #
+  # GenServer APIs
+  #
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  #
+  # GenServer behaviors
+  #
+
+  def init(opts) do
+    _ = Logger.info("Starting Ethereum height monitor.")
+    _ = install_alarm_handler()
+
+    state = %__MODULE__{
+      check_interval_ms: Keyword.fetch!(opts, :check_interval_ms),
+      stall_threshold_ms: Keyword.fetch!(opts, :stall_threshold_ms),
+      synced_at: DateTime.utc_now(),
+      eth_module: Keyword.fetch!(opts, :eth_module),
+      alarm_module: Keyword.fetch!(opts, :alarm_module),
+      event_bus: Keyword.fetch!(opts, :event_bus)
+    }
+
+    {:ok, tref} = :timer.send_after(state.check_interval_ms, :check_new_height)
```

Comment:
validating the assumption:
this means that the process, upon start, will not check for new block height. would it be better if the first check is done immediately?
```
{:ok, tref} = :timer.send_after(0, :check_new_height)
```
(not tested)